plugins/org.eclipse.elk.alg.vertiflex/src/org/eclipse/elk/alg/vertiflex/p2relative/RelativeXPlacer.java

```diff
@@ -0,0 +1,670 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.elk.alg.vertiflex.InternalProperties;
+import org.eclipse.elk.alg.vertiflex.VertiFlexLayoutPhases;
+import org.eclipse.elk.alg.vertiflex.options.VertiFlexOptions;
+import org.eclipse.elk.core.alg.ILayoutPhase;
+import org.eclipse.elk.core.alg.LayoutProcessorConfiguration;
+import org.eclipse.elk.core.math.ElkMargin;
+import org.eclipse.elk.core.options.CoreOptions;
+import org.eclipse.elk.core.util.IElkProgressMonitor;
+import org.eclipse.elk.core.util.Pair;
+import org.eclipse.elk.graph.ElkNode;
+
+/**
+ * Node placer that positions nodes horizontally using coordinates that are relative to parent nodes.
+ *
+ */
+public class RelativeXPlacer implements ILayoutPhase<VertiFlexLayoutPhases, ElkNode> {
+
+    private double spacingNodeNode;
+    private boolean considerNodeModelOrder;
+
+    // a constant for moving every Outline to a minimal y-pos
+    private static final double MINIMAL_Y = -100.0;
+    
+    @Override
+    public void process(final ElkNode graph, final IElkProgressMonitor progressMonitor) {
+
+        progressMonitor.begin("XPlacer", 1);
+        
+        spacingNodeNode = graph.getProperty(CoreOptions.SPACING_NODE_NODE);
+        considerNodeModelOrder = graph.getProperty(VertiFlexOptions.CONSIDER_NODE_MODEL_ORDER);
+        
+        if (!graph.getChildren().isEmpty()) {
+            ElkNode parent = graph.getProperty(InternalProperties.ROOT_NODE);
+            
+            switch (graph.getProperty(VertiFlexOptions.LAYOUT_STRATEGY)) {
+                case STRAIGHT:
+                    recursiveStraightlinePlacement(parent);
+                    break;
+                case BEND:
+                    recursiveBentlinePlacement(parent);
+                    break;
+                default:
+                    break;   
+            }
+        }
+
+        progressMonitor.done();
+    }
+    
+    /** Computes the distance between two outlines. */
+    private double outlineDistance(final OutlineNode outline1, final OutlineNode outline2) {
+        
+        OutlineNode changedOutline1 = new OutlineNode(outline1.getRelativeX(), MINIMAL_Y, 
+                new OutlineNode(0.0, outline1.getAbsoluteY(), outline1.getNext()));
+        OutlineNode changedOutline2 = new OutlineNode(outline2.getRelativeX(), MINIMAL_Y, 
+                new OutlineNode(0.0, outline2.getAbsoluteY(), outline2.getNext()));
+        
+        // the return value
+        double dist = changedOutline1.getRelativeX() - changedOutline2.getRelativeX();
+        
+        OutlineNode o1;
+        OutlineNode o2;
+        double x1;
+        double x2;
+        double deltaX, deltaY;
+        double newdist;
+        
+        
+        // first run (compare points of o1  with o2)
+        o1 = changedOutline1;
+        o2 = changedOutline2;
+        x1 = o1.getRelativeX();
+        x2 = o2.getRelativeX();
```

Comment:
This might be a problem, if the outline nodes already have positions from the previous run?